OpenRA.Mods.Common/Scripting/Global/DateTimeGlobal.cs

```diff
@@ -43,5 +49,39 @@ public int Minutes(int minutes)
 		{
 			return Seconds(minutes * 60);
 		}
+
+		[Desc("Return or set the time limit (in ticks). When setting, the time limit will count from now. Setting the time limit to 0 will disable it.")]
+		public int TimeLimit
+		{
+			get
+			{
+				return tlm != null ? tlm.TimeLimit : 0;
+			}
+
+			set
+			{
+				if (tlm != null)
+					tlm.TimeLimit = value == 0 ? 0 : value + GameTime;
+				else
+					throw new InvalidOperationException("Cannot set TimeLimit, TimeLimitManager trait is missing.");
```

Comment:
These should be `LuaException`s, which will trigger the ingame script-is-broken dialog instead of crashing the game.